src/pal/src/CMakeLists.txt

```diff
@@ -70,6 +63,16 @@ elseif(PAL_CMAKE_PLATFORM_ARCH_ARM64)
   add_definitions(-D_WIN64=1)
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL Linux AND NOT CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+  # Currently the _xstate is not available on Alpine Linux
+  add_definitions(-DXSTATE_SUPPORTED)
+endif(CMAKE_SYSTEM_NAME STREQUAL Linux AND NOT CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+
+if(CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+  # Setting RLIMIT_NOFILE breaks debugging of coreclr on Alpine Linux for some reason
```

Comment:
So, this is what happens:
```
999         result = getrlimit(RLIMIT_NOFILE, &rlp);
(gdb) n
1000        if (result != 0)
(gdb)
1006        rlp.rlim_cur = rlp.rlim_max;
(gdb) p/x rlp
$1 = {rlim_cur = 0x80000, rlim_max = 0x100000}
(gdb) n
1007        result = setrlimit(RLIMIT_NOFILE, &rlp);
(gdb)

Thread 3 "corerun" received signal ?, Unknown signal.
[Switching to LWP 28373]
0x00007ffff7da8087 in syscall () from /lib/ld-musl-x86_64.so.1
```
See - I've just stepped over the setrlimit. From that time on, I cannot do anything in GDB - the "c" doesn't work etc. Btw, the same thing happens even if I don't step through the code and just run it under GDB. 
If I run it out of the GDB, I haven't noticed any problem.
I've even tried to lower the rlim_max by 1 before assigning it to the rlim_cur, but it didn't help.
It is interesting that LLDB doesn't have this problem. 
Anyways, it seems that the default 524288 file descriptors are enough.